freeipa/src/main/resources/freeipa-salt/salt/freeipa/init.sls

```diff
@@ -17,17 +26,36 @@ net.ipv6.conf.lo.disable_ipv6:
     - mode: 700
     - source: salt://freeipa/scripts/freeipa_install.sh
 
+{% if salt['pillar.get']('freeipa:mode') == 'FIRST_INSTALL' %}
 install-freeipa:
   cmd.run:
-    - name: /opt/salt/scripts/freeipa_install.sh && echo $(date +%Y-%m-%d:%H:%M:%S) >> /var/log/freeipa_install-executed
+    - name: /opt/salt/scripts/freeipa_install.sh "FIRST_INSTALL" && echo $(date +%Y-%m-%d:%H:%M:%S) >> /var/log/freeipa_install-executed
```

Comment:
Could you evaluate solving this problem using grains instead of pillar?

Reply:
I updated it to use grains for the role assignment. I still have to see why the integration tests are failing (they don't run locally for me so I will see if I can get help on running it tomorrow).